src/PrestaShopBundle/Controller/Admin/AuthorizationController.php

```diff
@@ -0,0 +1,325 @@
+use PrestaShop\PrestaShop\Core\Domain\Exception\DomainConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\Exception\DomainException;
+use PrestaShop\PrestaShop\Core\Security\Exception\UnableToRenameAdminDirectoryException;
+use PrestaShopBundle\Form\Admin\Login\ForgotPasswordType;
+use PrestaShopBundle\Form\Admin\Login\LoginType;
+use PrestaShopBundle\Form\Admin\Login\ResetPasswordType;
+use PrestaShopBundle\Security\Annotation\DemoRestricted;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Component\Security\Core\Exception\BadCredentialsException;
+use Symfony\Component\Security\Core\Exception\UsernameNotFoundException;
+
+/**
+ * Class AuthorizationController responsible for employee login page.
+ */
+class AuthorizationController extends FrameworkBundleAdminController
+{
+    /**
+     * Render and handle login page.
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function loginAction(Request $request)
+    {
+        $authenticationUtils = $this->get('security.authentication_utils');
+        $authenticationError = $authenticationUtils->getLastAuthenticationError();
+        $loginForm = $this->createForm(LoginType::class, [
+            'redirect_url' => $request->query->get('redirect'),
+        ]);
+
+        if (null !== $authenticationError) {
+            $this->addFlash(
+                'error',
+                $this->getErrorMessageForException(
+                    $authenticationError,
+                    $this->getErrorMessages()
+                )
+            );
+        }
+
+        $forgotPasswordForm = $this->createForm(ForgotPasswordType::class);
+
+        return $this->renderLoginPage($request, [
+            'loginForm' => $loginForm->createView(),
+            'forgotPasswordForm' => $forgotPasswordForm->createView(),
+        ]);
+    }
+
+    /**
+     * Handle sending reset password link to the employee.
+     *
+     * @DemoRestricted(redirectRoute="_admin_login")
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function forgotPasswordAction(Request $request)
+    {
+        $loginForm = $this->createForm(LoginType::class);
+        $forgotPasswordForm = $this->createForm(ForgotPasswordType::class);
+        $forgotPasswordForm->handleRequest($request);
+        $showForgotPasswordForm = true;
+
+        if ($forgotPasswordForm->isSubmitted() && $forgotPasswordForm->isValid()) {
+            try {
+                $this->getCommandBus()->handle(
+                    new SendResetPasswordEmailCommand($forgotPasswordForm->getData()['email'])
+                );
+
+                $this->addFlash(
+                    'success',
+                    $this->trans(
+                        'Check your mailbox and click on the link to reset your password.',
```

Comment:
@LouiseBonnard do you think this message should be changed? IMO it's a bit too aggressive :slightly_smiling_face: